### PR TITLE
Improve docker build cost time

### PR DESCRIPTION
**Pull request by icezohu**
lower docker build cost time from 33m to 14m.

will hold on merge before deciding when to release.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,25 +1,30 @@
 FROM node:16-alpine AS BUILD_IMAGE
 
-RUN apk update && apk add --no-cache curl bash tini git docker-cli docker-compose grep make python3 g++
+# RUN apk update && apk add --no-cache curl bash tini git docker-cli docker-compose grep make python3 g++
```
Comment (mzxyz) [inline Dockerfile:3]: Have you test the local build docker image, seems it shouldn't work if missing `docker-cli`

Comment (icezohu) [reply Dockerfile:3]: yes, it's tested. there are two phases building two differrent images. the `BUILD_IMAGE` is only for building node modules. and the second image will install system packages including `docker-cli` which will actually take effect when it's working.

**Comment by HuberTRoy** [comment]
Please also change `readme` and `rush.json` for use yarn.

**Comment by icezohu** [comment]
> Please also change `readme` and `rush.json` for use yarn.

`pnpm to yarn` change is still in test and will raise another pr for it, if it's ok.

**Comment by icezohu** [comment]
this pr is included in pr #196 . now close.

**Pull request closed without merging**